### PR TITLE
added auto-init settings

### PR DESCRIPTION
**Pull request by iruzevic**
# Description

- added auto-init setting in admin.
- fixed form init to correspond to that settings.
- moved the part of the window object to be loaded every-time.
- added bubbling to custom event
- updating docs
- 

# Screenshots / Videos
Screenshot 2022-02-03 at 12 32 58

### CODE REVIEW

**Review by mbmjertan — APPROVED** [review]
Looks good

File: src/Settings/Settings/SettingsGeneral.php [hunk 1]
```diff
@@ -164,6 +165,7 @@ public function getSettingsGlobalData(): array
 					Keep in mind that you know what you are doing and that you will provide your own styles and scripts.<br /><br />
 					Disabling styles will remove all of plugin styles for frontend and block editor.<br /><br />
 					Disabling scripts will remove all scripts on the front end. Keep in mind this will make the frontend validation and submitting the form stop working. <br /><br />
+					Disabling scripts auto-init will load all JS but will not initialize it, you have to doit mainually. Please refer the documentation of the examples. <br /><br />
```
Comment (iobrado) [inline src/Settings/Settings/SettingsGeneral.php:168]:
```suggestion
					Disabling scripts auto-init will load all JS but will not initialize it, you have to do it mainually. Please refer the documentation of the examples. <br /><br />
```

**Review by iobrado — APPROVED** [review]
Found just one typo 🙂   otherwise looks good

File: src/Settings/Settings/SettingsGeneral.php [hunk 2]
```diff
@@ -164,6 +165,7 @@ public function getSettingsGlobalData(): array
 					Keep in mind that you know what you are doing and that you will provide your own styles and scripts.<br /><br />
 					Disabling styles will remove all of plugin styles for frontend and block editor.<br /><br />
 					Disabling scripts will remove all scripts on the front end. Keep in mind this will make the frontend validation and submitting the form stop working. <br /><br />
+					Disabling scripts auto-init will load all JS but will not initialize it, you have to do it mainually. Please refer the documentation of the examples. <br /><br />
```
Comment (dadadavorin) [inline src/Settings/Settings/SettingsGeneral.php:168]:
```suggestion
					Disabling scripts auto-init will load all JS but will not initialize it, you have to do it mainually. Please refer to the documentation of the examples. <br /><br />
```

**Review by dadadavorin — APPROVED** [review]

File: src/Blocks/components/form/assets/form.js [hunk 3]
```diff
@@ -649,7 +650,9 @@ export class Form {
 
 	// Dispatch custom event.
 	dispatchFormEvent(element, name) {
-		const event = new CustomEvent(`esForms${name}`);
+		const event = new CustomEvent(`esForms${name}`, {
+			bubbles: true
+		});
```
Comment (volfkarlo) [inline src/Blocks/components/form/assets/form.js:655]: do we have the ensurance this will dispatch after all listeners are set?

**Review by volfkarlo — CHANGES_REQUESTED** [review]

File: src/Blocks/components/form/assets/form.js (hunk 3 above)
Comment (iruzevic) [reply src/Blocks/components/form/assets/form.js:655]: yes if you put the correct listeners it will

**Review by goranalkovic-infinum — APPROVED** [review]

**Pull request merged**